File: 프로그래머스/0/120860. 직사각형 넓이 구하기/직사각형 넓이 구하기.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int solution(vector<vector<int>> dots) {
    // x좌표들의 최대값, 최소값을 구하기 위한 벡터
    vector<int> x_coords;
    // y좌표들의 최대값, 최소값을 구하기 위한 벡터
    vector<int> y_coords;
    
    // 각 점의 x좌표와 y좌표를 분리하여 저장
    for(const auto& dot : dots) {
        x_coords.push_back(dot[0]);
        y_coords.push_back(dot[1]);
    }
    
    // 가로 길이 = x좌표 최대값 - 최소값
    int width = *max_element(x_coords.begin(), x_coords.end()) - 
                *min_element(x_coords.begin(), x_coords.end());
    
    // 세로 길이 = y좌표 최대값 - 최소값
    int height = *max_element(y_coords.begin(), y_coords.end()) - 
                 *min_element(y_coords.begin(), y_coords.end());
    
    // 넓이 = 가로 × 세로
    return width * height;
}

File: 프로그래머스/0/120860. 직사각형 넓이 구하기/직사각형 넓이 구하기.swift
import Foundation

func solution(_ dots:[[Int]]) -> Int {
    // let xValues = dots.map { $0[0] }
    // let yValues = dots.map { $0[1] }
    // let width = xValues.max()! - xValues.min()!
    // let height = yValues.max()! - yValues.min()!
    // return width * height
    
    // tuple sort (x와 y 정렬)
    let dots = dots.sorted(by: { ($0[0], $0[1]) < ($1[0], $1[1]) })
    return (dots.last![0] - dots.first![0]) * (dots.last![1] - dots.first![1])
}